packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/TxDetailModal/AdvancedTxDetails/AmountDetails.tsx

```diff
@@ -47,10 +52,26 @@ interface AmountDetailsProps {
 
 // TODO: Do not show FEE for sent but not mine transactions
 export const AmountDetails = ({ tx, isTestnet }: AmountDetailsProps) => {
+    const fiatCurrencyCode = useSelector(selectLocalCurrency);
+    const cryptoFiatRateKey = getFiatRateKey(tx.symbol, fiatCurrencyCode);
+    const tokenFiatRateKey = getFiatRateKey(
+        tx.symbol,
+        fiatCurrencyCode,
+        tx.tokens[0]?.contract as TokenAddress,
+    );
+
+    const historicCryptoRate = useSelector(state =>
+        selectHistoricFiatRatesByTimestamp(state, cryptoFiatRateKey, tx.blockTime as Timestamp),
```

Comment:
I know its maybe out-of-scope, but would it be possible (in other PR) introduce those type to blockchin-link?

Reply:
So we can avoid this heavy type-casting

Reply:
It would be nice but let's not making this bigger than it already is